// @aws-sdk/client-bedrock-agent-runtime/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  ActionGroupInvocationInputFilterSensitiveLog: () => ActionGroupInvocationInputFilterSensitiveLog,
  ActionGroupInvocationOutputFilterSensitiveLog: () => ActionGroupInvocationOutputFilterSensitiveLog,
  ActionInvocationType: () => ActionInvocationType,
  ApiInvocationInputFilterSensitiveLog: () => ApiInvocationInputFilterSensitiveLog,
  ApiResultFilterSensitiveLog: () => ApiResultFilterSensitiveLog,
  AttributionFilterSensitiveLog: () => AttributionFilterSensitiveLog,
  BadGatewayException: () => BadGatewayException,
  BedrockAgentRuntime: () => BedrockAgentRuntime,
  BedrockAgentRuntimeClient: () => BedrockAgentRuntimeClient,
  BedrockAgentRuntimeServiceException: () => BedrockAgentRuntimeServiceException,
  ByteContentDocFilterSensitiveLog: () => ByteContentDocFilterSensitiveLog,
  ByteContentFileFilterSensitiveLog: () => ByteContentFileFilterSensitiveLog,
  CitationFilterSensitiveLog: () => CitationFilterSensitiveLog,
  ConfirmationState: () => ConfirmationState,
  ConflictException: () => ConflictException,
  CreationMode: () => CreationMode,
  DeleteAgentMemoryCommand: () => DeleteAgentMemoryCommand,
  DependencyFailedException: () => DependencyFailedException,
  ExecutionType: () => ExecutionType,
  ExternalSourceFilterSensitiveLog: () => ExternalSourceFilterSensitiveLog,
  ExternalSourceType: () => ExternalSourceType,
  ExternalSourcesGenerationConfigurationFilterSensitiveLog: () => ExternalSourcesGenerationConfigurationFilterSensitiveLog,
  ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog: () => ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog,
  FailureTraceFilterSensitiveLog: () => FailureTraceFilterSensitiveLog,
  FilePartFilterSensitiveLog: () => FilePartFilterSensitiveLog,
  FileSourceFilterSensitiveLog: () => FileSourceFilterSensitiveLog,
  FileSourceType: () => FileSourceType,
  FileUseCase: () => FileUseCase,
  FinalResponseFilterSensitiveLog: () => FinalResponseFilterSensitiveLog,
  FlowCompletionEventFilterSensitiveLog: () => FlowCompletionEventFilterSensitiveLog,
  FlowCompletionReason: () => FlowCompletionReason,
  FlowInputContent: () => FlowInputContent,
  FlowInputContentFilterSensitiveLog: () => FlowInputContentFilterSensitiveLog,
  FlowInputFilterSensitiveLog: () => FlowInputFilterSensitiveLog,
  FlowOutputContent: () => FlowOutputContent,
  FlowOutputEventFilterSensitiveLog: () => FlowOutputEventFilterSensitiveLog,
  FlowResponseStream: () => FlowResponseStream,
  FlowResponseStreamFilterSensitiveLog: () => FlowResponseStreamFilterSensitiveLog,
  GeneratedResponsePartFilterSensitiveLog: () => GeneratedResponsePartFilterSensitiveLog,
  GenerationConfigurationFilterSensitiveLog: () => GenerationConfigurationFilterSensitiveLog,
  GetAgentMemoryCommand: () => GetAgentMemoryCommand,
  GuadrailAction: () => GuadrailAction,
  GuardrailAction: () => GuardrailAction,
  GuardrailAssessmentFilterSensitiveLog: () => GuardrailAssessmentFilterSensitiveLog,
  GuardrailContentFilterConfidence: () => GuardrailContentFilterConfidence,
  GuardrailContentFilterFilterSensitiveLog: () => GuardrailContentFilterFilterSensitiveLog,
  GuardrailContentFilterType: () => GuardrailContentFilterType,
  GuardrailContentPolicyAction: () => GuardrailContentPolicyAction,
  GuardrailContentPolicyAssessmentFilterSensitiveLog: () => GuardrailContentPolicyAssessmentFilterSensitiveLog,
  GuardrailCustomWordFilterSensitiveLog: () => GuardrailCustomWordFilterSensitiveLog,
  GuardrailManagedWordFilterSensitiveLog: () => GuardrailManagedWordFilterSensitiveLog,
  GuardrailManagedWordType: () => GuardrailManagedWordType,
  GuardrailPiiEntityFilterFilterSensitiveLog: () => GuardrailPiiEntityFilterFilterSensitiveLog,
  GuardrailPiiEntityType: () => GuardrailPiiEntityType,
  GuardrailRegexFilterFilterSensitiveLog: () => GuardrailRegexFilterFilterSensitiveLog,
  GuardrailSensitiveInformationPolicyAction: () => GuardrailSensitiveInformationPolicyAction,
  GuardrailSensitiveInformationPolicyAssessmentFilterSensitiveLog: () => GuardrailSensitiveInformationPolicyAssessmentFilterSensitiveLog,
  GuardrailTopicFilterSensitiveLog: () => GuardrailTopicFilterSensitiveLog,
  GuardrailTopicPolicyAction: () => GuardrailTopicPolicyAction,
  GuardrailTopicPolicyAssessmentFilterSensitiveLog: () => GuardrailTopicPolicyAssessmentFilterSensitiveLog,
  GuardrailTopicType: () => GuardrailTopicType,
  GuardrailTraceFilterSensitiveLog: () => GuardrailTraceFilterSensitiveLog,
  GuardrailWordPolicyAction: () => GuardrailWordPolicyAction,
  GuardrailWordPolicyAssessmentFilterSensitiveLog: () => GuardrailWordPolicyAssessmentFilterSensitiveLog,
  InputFileFilterSensitiveLog: () => InputFileFilterSensitiveLog,
  InternalServerException: () => InternalServerException,
  InvocationInputFilterSensitiveLog: () => InvocationInputFilterSensitiveLog,
  InvocationInputMember: () => InvocationInputMember,
  InvocationInputMemberFilterSensitiveLog: () => InvocationInputMemberFilterSensitiveLog,
  InvocationResultMember: () => InvocationResultMember,
  InvocationResultMemberFilterSensitiveLog: () => InvocationResultMemberFilterSensitiveLog,
  InvocationType: () => InvocationType,
  InvokeAgentCommand: () => InvokeAgentCommand,
  InvokeAgentRequestFilterSensitiveLog: () => InvokeAgentRequestFilterSensitiveLog,
  InvokeAgentResponseFilterSensitiveLog: () => InvokeAgentResponseFilterSensitiveLog,
  InvokeFlowCommand: () => InvokeFlowCommand,
  InvokeFlowRequestFilterSensitiveLog: () => InvokeFlowRequestFilterSensitiveLog,
  InvokeFlowResponseFilterSensitiveLog: () => InvokeFlowResponseFilterSensitiveLog,
  KnowledgeBaseConfigurationFilterSensitiveLog: () => KnowledgeBaseConfigurationFilterSensitiveLog,
  KnowledgeBaseLookupInputFilterSensitiveLog: () => KnowledgeBaseLookupInputFilterSensitiveLog,
  KnowledgeBaseLookupOutputFilterSensitiveLog: () => KnowledgeBaseLookupOutputFilterSensitiveLog,
  KnowledgeBaseQueryFilterSensitiveLog: () => KnowledgeBaseQueryFilterSensitiveLog,
  KnowledgeBaseRetrievalConfigurationFilterSensitiveLog: () => KnowledgeBaseRetrievalConfigurationFilterSensitiveLog,
  KnowledgeBaseRetrievalResultFilterSensitiveLog: () => KnowledgeBaseRetrievalResultFilterSensitiveLog,
  KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog: () => KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog,
  KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog: () => KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog,
  Memory: () => Memory,
  MemoryType: () => MemoryType,
  MetadataFilterSensitiveLog: () => MetadataFilterSensitiveLog,
  ModelInvocationInputFilterSensitiveLog: () => ModelInvocationInputFilterSensitiveLog,
  NodeType: () => NodeType,
  ObservationFilterSensitiveLog: () => ObservationFilterSensitiveLog,
  OrchestrationConfigurationFilterSensitiveLog: () => OrchestrationConfigurationFilterSensitiveLog,
  OrchestrationModelInvocationOutputFilterSensitiveLog: () => OrchestrationModelInvocationOutputFilterSensitiveLog,
  OrchestrationTrace: () => OrchestrationTrace,
  OrchestrationTraceFilterSensitiveLog: () => OrchestrationTraceFilterSensitiveLog,
  OutputFileFilterSensitiveLog: () => OutputFileFilterSensitiveLog,
  PayloadPartFilterSensitiveLog: () => PayloadPartFilterSensitiveLog,
  PostProcessingModelInvocationOutputFilterSensitiveLog: () => PostProcessingModelInvocationOutputFilterSensitiveLog,
  PostProcessingParsedResponseFilterSensitiveLog: () => PostProcessingParsedResponseFilterSensitiveLog,
  PostProcessingTrace: () => PostProcessingTrace,
  PostProcessingTraceFilterSensitiveLog: () => PostProcessingTraceFilterSensitiveLog,
  PreProcessingModelInvocationOutputFilterSensitiveLog: () => PreProcessingModelInvocationOutputFilterSensitiveLog,
  PreProcessingParsedResponseFilterSensitiveLog: () => PreProcessingParsedResponseFilterSensitiveLog,
  PreProcessingTrace: () => PreProcessingTrace,
  PreProcessingTraceFilterSensitiveLog: () => PreProcessingTraceFilterSensitiveLog,
  PromptTemplateFilterSensitiveLog: () => PromptTemplateFilterSensitiveLog,
  PromptType: () => PromptType,
  QueryTransformationType: () => QueryTransformationType,
  RationaleFilterSensitiveLog: () => RationaleFilterSensitiveLog,
  RawResponseFilterSensitiveLog: () => RawResponseFilterSensitiveLog,
  RepromptResponseFilterSensitiveLog: () => RepromptResponseFilterSensitiveLog,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResponseState: () => ResponseState,
  ResponseStream: () => ResponseStream,
  ResponseStreamFilterSensitiveLog: () => ResponseStreamFilterSensitiveLog,
  RetrievalFilter: () => RetrievalFilter,
  RetrievalFilterFilterSensitiveLog: () => RetrievalFilterFilterSensitiveLog,
  RetrievalResultContentFilterSensitiveLog: () => RetrievalResultContentFilterSensitiveLog,
  RetrievalResultLocationFilterSensitiveLog: () => RetrievalResultLocationFilterSensitiveLog,
  RetrievalResultLocationType: () => RetrievalResultLocationType,
  RetrieveAndGenerateCommand: () => RetrieveAndGenerateCommand,
  RetrieveAndGenerateConfigurationFilterSensitiveLog: () => RetrieveAndGenerateConfigurationFilterSensitiveLog,
  RetrieveAndGenerateInputFilterSensitiveLog: () => RetrieveAndGenerateInputFilterSensitiveLog,
  RetrieveAndGenerateOutputFilterSensitiveLog: () => RetrieveAndGenerateOutputFilterSensitiveLog,
  RetrieveAndGenerateRequestFilterSensitiveLog: () => RetrieveAndGenerateRequestFilterSensitiveLog,
  RetrieveAndGenerateResponseFilterSensitiveLog: () => RetrieveAndGenerateResponseFilterSensitiveLog,
  RetrieveAndGenerateType: () => RetrieveAndGenerateType,
  RetrieveCommand: () => RetrieveCommand,
  RetrieveRequestFilterSensitiveLog: () => RetrieveRequestFilterSensitiveLog,
  RetrieveResponseFilterSensitiveLog: () => RetrieveResponseFilterSensitiveLog,
  RetrievedReferenceFilterSensitiveLog: () => RetrievedReferenceFilterSensitiveLog,
  ReturnControlPayloadFilterSensitiveLog: () => ReturnControlPayloadFilterSensitiveLog,
  SearchType: () => SearchType,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  SessionStateFilterSensitiveLog: () => SessionStateFilterSensitiveLog,
  Source: () => Source,
  TextResponsePartFilterSensitiveLog: () => TextResponsePartFilterSensitiveLog,
  ThrottlingException: () => ThrottlingException,
  Trace: () => Trace,
  TraceFilterSensitiveLog: () => TraceFilterSensitiveLog,
  TracePartFilterSensitiveLog: () => TracePartFilterSensitiveLog,
  Type: () => Type,
  UsageFilterSensitiveLog: () => UsageFilterSensitiveLog,
  ValidationException: () => ValidationException,
  __Client: () => import_smithy_client.Client,
  paginateGetAgentMemory: () => paginateGetAgentMemory,
  paginateRetrieve: () => paginateRetrieve
});
module.exports = __toCommonJS(src_exports);

// src/BedrockAgentRuntimeClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_eventstream_serde_config_resolver = require("@smithy/eventstream-serde-config-resolver");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "bedrock"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/BedrockAgentRuntimeClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/BedrockAgentRuntimeClient.ts
var _BedrockAgentRuntimeClient = class _BedrockAgentRuntimeClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_1);
    const _config_3 = (0, import_middleware_retry.resolveRetryConfig)(_config_2);
    const _config_4 = (0, import_config_resolver.resolveRegionConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_5);
    const _config_7 = (0, import_eventstream_serde_config_resolver.resolveEventStreamSerdeConfig)(_config_6);
    const _config_8 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_7);
    const _config_9 = resolveRuntimeExtensions(_config_8, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_9);
    this.config = _config_9;
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: import_httpAuthSchemeProvider.defaultBedrockAgentRuntimeHttpAuthSchemeParametersProvider,
        identityProviderConfigProvider: async (config) => new import_core.DefaultIdentityProviderConfig({
          "aws.auth#sigv4": config.credentials
        })
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
};
__name(_BedrockAgentRuntimeClient, "BedrockAgentRuntimeClient");
var BedrockAgentRuntimeClient = _BedrockAgentRuntimeClient;

// src/BedrockAgentRuntime.ts


// src/commands/DeleteAgentMemoryCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");


// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/BedrockAgentRuntimeServiceException.ts

var _BedrockAgentRuntimeServiceException = class _BedrockAgentRuntimeServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _BedrockAgentRuntimeServiceException.prototype);
  }
};
__name(_BedrockAgentRuntimeServiceException, "BedrockAgentRuntimeServiceException");
var BedrockAgentRuntimeServiceException = _BedrockAgentRuntimeServiceException;

// src/models/models_0.ts

var _AccessDeniedException = class _AccessDeniedException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var ExecutionType = {
  LAMBDA: "LAMBDA",
  RETURN_CONTROL: "RETURN_CONTROL"
};
var ActionInvocationType = {
  RESULT: "RESULT",
  USER_CONFIRMATION: "USER_CONFIRMATION",
  USER_CONFIRMATION_AND_RESULT: "USER_CONFIRMATION_AND_RESULT"
};
var _BadGatewayException = class _BadGatewayException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "BadGatewayException",
      $fault: "server",
      ...opts
    });
    this.name = "BadGatewayException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _BadGatewayException.prototype);
    this.resourceName = opts.resourceName;
  }
};
__name(_BadGatewayException, "BadGatewayException");
var BadGatewayException = _BadGatewayException;
var _ConflictException = class _ConflictException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _DependencyFailedException = class _DependencyFailedException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DependencyFailedException",
      $fault: "client",
      ...opts
    });
    this.name = "DependencyFailedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DependencyFailedException.prototype);
    this.resourceName = opts.resourceName;
  }
};
__name(_DependencyFailedException, "DependencyFailedException");
var DependencyFailedException = _DependencyFailedException;
var _InternalServerException = class _InternalServerException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var FlowInputContent;
((FlowInputContent2) => {
  FlowInputContent2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.document !== void 0)
      return visitor.document(value.document);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(FlowInputContent || (FlowInputContent = {}));
var FlowCompletionReason = {
  SUCCESS: "SUCCESS"
};
var FlowOutputContent;
((FlowOutputContent3) => {
  FlowOutputContent3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.document !== void 0)
      return visitor.document(value.document);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(FlowOutputContent || (FlowOutputContent = {}));
var NodeType = {
  CONDITION_NODE: "ConditionNode",
  FLOW_INPUT_NODE: "FlowInputNode",
  FLOW_OUTPUT_NODE: "FlowOutputNode",
  KNOWLEDGE_BASE_NODE: "KnowledgeBaseNode",
  LAMBDA_FUNCTION_NODE: "LambdaFunctionNode",
  LEX_NODE: "LexNode",
  PROMPT_NODE: "PromptNode"
};
var _ResourceNotFoundException = class _ResourceNotFoundException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var _ThrottlingException = class _ThrottlingException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var _ValidationException = class _ValidationException extends BedrockAgentRuntimeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var FlowResponseStream;
((FlowResponseStream3) => {
  FlowResponseStream3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.flowOutputEvent !== void 0)
      return visitor.flowOutputEvent(value.flowOutputEvent);
    if (value.flowCompletionEvent !== void 0)
      return visitor.flowCompletionEvent(value.flowCompletionEvent);
    if (value.internalServerException !== void 0)
      return visitor.internalServerException(value.internalServerException);
    if (value.validationException !== void 0)
      return visitor.validationException(value.validationException);
    if (value.resourceNotFoundException !== void 0)
      return visitor.resourceNotFoundException(value.resourceNotFoundException);
    if (value.serviceQuotaExceededException !== void 0)
      return visitor.serviceQuotaExceededException(value.serviceQuotaExceededException);
    if (value.throttlingException !== void 0)
      return visitor.throttlingException(value.throttlingException);
    if (value.accessDeniedException !== void 0)
      return visitor.accessDeniedException(value.accessDeniedException);
    if (value.conflictException !== void 0)
      return visitor.conflictException(value.conflictException);
    if (value.dependencyFailedException !== void 0)
      return visitor.dependencyFailedException(value.dependencyFailedException);
    if (value.badGatewayException !== void 0)
      return visitor.badGatewayException(value.badGatewayException);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(FlowResponseStream || (FlowResponseStream = {}));
var FileSourceType = {
  BYTE_CONTENT: "BYTE_CONTENT",
  S3: "S3"
};
var FileUseCase = {
  CHAT: "CHAT",
  CODE_INTERPRETER: "CODE_INTERPRETER"
};
var SearchType = {
  HYBRID: "HYBRID",
  SEMANTIC: "SEMANTIC"
};
var ConfirmationState = {
  CONFIRM: "CONFIRM",
  DENY: "DENY"
};
var ResponseState = {
  FAILURE: "FAILURE",
  REPROMPT: "REPROMPT"
};
var InvocationResultMember;
((InvocationResultMember3) => {
  InvocationResultMember3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.apiResult !== void 0)
      return visitor.apiResult(value.apiResult);
    if (value.functionResult !== void 0)
      return visitor.functionResult(value.functionResult);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(InvocationResultMember || (InvocationResultMember = {}));
var RetrievalResultLocationType = {
  CONFLUENCE: "CONFLUENCE",
  S3: "S3",
  SALESFORCE: "SALESFORCE",
  SHAREPOINT: "SHAREPOINT",
  WEB: "WEB"
};
var InvocationInputMember;
((InvocationInputMember2) => {
  InvocationInputMember2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.apiInvocationInput !== void 0)
      return visitor.apiInvocationInput(value.apiInvocationInput);
    if (value.functionInvocationInput !== void 0)
      return visitor.functionInvocationInput(value.functionInvocationInput);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(InvocationInputMember || (InvocationInputMember = {}));
var GuardrailAction = {
  INTERVENED: "INTERVENED",
  NONE: "NONE"
};
var GuardrailContentPolicyAction = {
  BLOCKED: "BLOCKED"
};
var GuardrailContentFilterConfidence = {
  HIGH: "HIGH",
  LOW: "LOW",
  MEDIUM: "MEDIUM",
  NONE: "NONE"
};
var GuardrailContentFilterType = {
  HATE: "HATE",
  INSULTS: "INSULTS",
  MISCONDUCT: "MISCONDUCT",
  PROMPT_ATTACK: "PROMPT_ATTACK",
  SEXUAL: "SEXUAL",
  VIOLENCE: "VIOLENCE"
};
var GuardrailSensitiveInformationPolicyAction = {
  ANONYMIZED: "ANONYMIZED",
  BLOCKED: "BLOCKED"
};
var GuardrailPiiEntityType = {
  ADDRESS: "ADDRESS",
  AGE: "AGE",
  AWS_ACCESS_KEY: "AWS_ACCESS_KEY",
  AWS_SECRET_KEY: "AWS_SECRET_KEY",
  CA_HEALTH_NUMBER: "CA_HEALTH_NUMBER",
  CA_SOCIAL_INSURANCE_NUMBER: "CA_SOCIAL_INSURANCE_NUMBER",
  CREDIT_DEBIT_CARD_CVV: "CREDIT_DEBIT_CARD_CVV",
  CREDIT_DEBIT_CARD_EXPIRY: "CREDIT_DEBIT_CARD_EXPIRY",
  CREDIT_DEBIT_CARD_NUMBER: "CREDIT_DEBIT_CARD_NUMBER",
  DRIVER_ID: "DRIVER_ID",
  EMAIL: "EMAIL",
  INTERNATIONAL_BANK_ACCOUNT_NUMBER: "INTERNATIONAL_BANK_ACCOUNT_NUMBER",
  IP_ADDRESS: "IP_ADDRESS",
  LICENSE_PLATE: "LICENSE_PLATE",
  MAC_ADDRESS: "MAC_ADDRESS",
  NAME: "NAME",
  PASSWORD: "PASSWORD",
  PHONE: "PHONE",
  PIN: "PIN",
  SWIFT_CODE: "SWIFT_CODE",
  UK_NATIONAL_HEALTH_SERVICE_NUMBER: "UK_NATIONAL_HEALTH_SERVICE_NUMBER",
  UK_NATIONAL_INSURANCE_NUMBER: "UK_NATIONAL_INSURANCE_NUMBER",
  UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER: "UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER",
  URL: "URL",
  USERNAME: "USERNAME",
  US_BANK_ACCOUNT_NUMBER: "US_BANK_ACCOUNT_NUMBER",
  US_BANK_ROUTING_NUMBER: "US_BANK_ROUTING_NUMBER",
  US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER: "US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER",
  US_PASSPORT_NUMBER: "US_PASSPORT_NUMBER",
  US_SOCIAL_SECURITY_NUMBER: "US_SOCIAL_SECURITY_NUMBER",
  VEHICLE_IDENTIFICATION_NUMBER: "VEHICLE_IDENTIFICATION_NUMBER"
};
var GuardrailTopicPolicyAction = {
  BLOCKED: "BLOCKED"
};
var GuardrailTopicType = {
  DENY: "DENY"
};
var GuardrailWordPolicyAction = {
  BLOCKED: "BLOCKED"
};
var GuardrailManagedWordType = {
  PROFANITY: "PROFANITY"
};
var InvocationType = {
  ACTION_GROUP: "ACTION_GROUP",
  ACTION_GROUP_CODE_INTERPRETER: "ACTION_GROUP_CODE_INTERPRETER",
  FINISH: "FINISH",
  KNOWLEDGE_BASE: "KNOWLEDGE_BASE"
};
var CreationMode = {
  DEFAULT: "DEFAULT",
  OVERRIDDEN: "OVERRIDDEN"
};
var PromptType = {
  KNOWLEDGE_BASE_RESPONSE_GENERATION: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
  ORCHESTRATION: "ORCHESTRATION",
  POST_PROCESSING: "POST_PROCESSING",
  PRE_PROCESSING: "PRE_PROCESSING"
};
var Source = {
  ACTION_GROUP: "ACTION_GROUP",
  KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
  PARSER: "PARSER"
};
var Type = {
  ACTION_GROUP: "ACTION_GROUP",
  ASK_USER: "ASK_USER",
  FINISH: "FINISH",
  KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
  REPROMPT: "REPROMPT"
};
var OrchestrationTrace;
((OrchestrationTrace3) => {
  OrchestrationTrace3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.rationale !== void 0)
      return visitor.rationale(value.rationale);
    if (value.invocationInput !== void 0)
      return visitor.invocationInput(value.invocationInput);
    if (value.observation !== void 0)
      return visitor.observation(value.observation);
    if (value.modelInvocationInput !== void 0)
      return visitor.modelInvocationInput(value.modelInvocationInput);
    if (value.modelInvocationOutput !== void 0)
      return visitor.modelInvocationOutput(value.modelInvocationOutput);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(OrchestrationTrace || (OrchestrationTrace = {}));
var PostProcessingTrace;
((PostProcessingTrace3) => {
  PostProcessingTrace3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.modelInvocationInput !== void 0)
      return visitor.modelInvocationInput(value.modelInvocationInput);
    if (value.modelInvocationOutput !== void 0)
      return visitor.modelInvocationOutput(value.modelInvocationOutput);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(PostProcessingTrace || (PostProcessingTrace = {}));
var PreProcessingTrace;
((PreProcessingTrace3) => {
  PreProcessingTrace3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.modelInvocationInput !== void 0)
      return visitor.modelInvocationInput(value.modelInvocationInput);
    if (value.modelInvocationOutput !== void 0)
      return visitor.modelInvocationOutput(value.modelInvocationOutput);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(PreProcessingTrace || (PreProcessingTrace = {}));
var Trace;
((Trace3) => {
  Trace3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.guardrailTrace !== void 0)
      return visitor.guardrailTrace(value.guardrailTrace);
    if (value.preProcessingTrace !== void 0)
      return visitor.preProcessingTrace(value.preProcessingTrace);
    if (value.orchestrationTrace !== void 0)
      return visitor.orchestrationTrace(value.orchestrationTrace);
    if (value.postProcessingTrace !== void 0)
      return visitor.postProcessingTrace(value.postProcessingTrace);
    if (value.failureTrace !== void 0)
      return visitor.failureTrace(value.failureTrace);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(Trace || (Trace = {}));
var ResponseStream;
((ResponseStream3) => {
  ResponseStream3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.chunk !== void 0)
      return visitor.chunk(value.chunk);
    if (value.trace !== void 0)
      return visitor.trace(value.trace);
    if (value.returnControl !== void 0)
      return visitor.returnControl(value.returnControl);
    if (value.internalServerException !== void 0)
      return visitor.internalServerException(value.internalServerException);
    if (value.validationException !== void 0)
      return visitor.validationException(value.validationException);
    if (value.resourceNotFoundException !== void 0)
      return visitor.resourceNotFoundException(value.resourceNotFoundException);
    if (value.serviceQuotaExceededException !== void 0)
      return visitor.serviceQuotaExceededException(value.serviceQuotaExceededException);
    if (value.throttlingException !== void 0)
      return visitor.throttlingException(value.throttlingException);
    if (value.accessDeniedException !== void 0)
      return visitor.accessDeniedException(value.accessDeniedException);
    if (value.conflictException !== void 0)
      return visitor.conflictException(value.conflictException);
    if (value.dependencyFailedException !== void 0)
      return visitor.dependencyFailedException(value.dependencyFailedException);
    if (value.badGatewayException !== void 0)
      return visitor.badGatewayException(value.badGatewayException);
    if (value.files !== void 0)
      return visitor.files(value.files);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(ResponseStream || (ResponseStream = {}));
var MemoryType = {
  SESSION_SUMMARY: "SESSION_SUMMARY"
};
var Memory;
((Memory3) => {
  Memory3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.sessionSummary !== void 0)
      return visitor.sessionSummary(value.sessionSummary);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(Memory || (Memory = {}));
var ExternalSourceType = {
  BYTE_CONTENT: "BYTE_CONTENT",
  S3: "S3"
};
var QueryTransformationType = {
  QUERY_DECOMPOSITION: "QUERY_DECOMPOSITION"
};
var RetrieveAndGenerateType = {
  EXTERNAL_SOURCES: "EXTERNAL_SOURCES",
  KNOWLEDGE_BASE: "KNOWLEDGE_BASE"
};
var GuadrailAction = {
  INTERVENED: "INTERVENED",
  NONE: "NONE"
};
var RetrievalFilter;
((RetrievalFilter2) => {
  RetrievalFilter2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.equals !== void 0)
      return visitor.equals(value.equals);
    if (value.notEquals !== void 0)
      return visitor.notEquals(value.notEquals);
    if (value.greaterThan !== void 0)
      return visitor.greaterThan(value.greaterThan);
    if (value.greaterThanOrEquals !== void 0)
      return visitor.greaterThanOrEquals(value.greaterThanOrEquals);
    if (value.lessThan !== void 0)
      return visitor.lessThan(value.lessThan);
    if (value.lessThanOrEquals !== void 0)
      return visitor.lessThanOrEquals(value.lessThanOrEquals);
    if (value.in !== void 0)
      return visitor.in(value.in);
    if (value.notIn !== void 0)
      return visitor.notIn(value.notIn);
    if (value.startsWith !== void 0)
      return visitor.startsWith(value.startsWith);
    if (value.listContains !== void 0)
      return visitor.listContains(value.listContains);
    if (value.stringContains !== void 0)
      return visitor.stringContains(value.stringContains);
    if (value.andAll !== void 0)
      return visitor.andAll(value.andAll);
    if (value.orAll !== void 0)
      return visitor.orAll(value.orAll);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(RetrievalFilter || (RetrievalFilter = {}));
var ActionGroupInvocationInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.actionGroupName && { actionGroupName: import_smithy_client.SENSITIVE_STRING },
  ...obj.verb && { verb: import_smithy_client.SENSITIVE_STRING },
  ...obj.apiPath && { apiPath: import_smithy_client.SENSITIVE_STRING },
  ...obj.function && { function: import_smithy_client.SENSITIVE_STRING }
}), "ActionGroupInvocationInputFilterSensitiveLog");
var ActionGroupInvocationOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING }
}), "ActionGroupInvocationOutputFilterSensitiveLog");
var FlowInputContentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.document !== void 0)
    return { document: obj.document };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "FlowInputContentFilterSensitiveLog");
var FlowInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.content && { content: import_smithy_client.SENSITIVE_STRING }
}), "FlowInputFilterSensitiveLog");
var InvokeFlowRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.inputs && { inputs: obj.inputs.map((item) => FlowInputFilterSensitiveLog(item)) }
}), "InvokeFlowRequestFilterSensitiveLog");
var FlowCompletionEventFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "FlowCompletionEventFilterSensitiveLog");
var FlowOutputEventFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.content && { content: obj.content }
}), "FlowOutputEventFilterSensitiveLog");
var FlowResponseStreamFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.flowOutputEvent !== void 0)
    return { flowOutputEvent: import_smithy_client.SENSITIVE_STRING };
  if (obj.flowCompletionEvent !== void 0)
    return { flowCompletionEvent: import_smithy_client.SENSITIVE_STRING };
  if (obj.internalServerException !== void 0)
    return { internalServerException: obj.internalServerException };
  if (obj.validationException !== void 0)
    return { validationException: obj.validationException };
  if (obj.resourceNotFoundException !== void 0)
    return { resourceNotFoundException: obj.resourceNotFoundException };
  if (obj.serviceQuotaExceededException !== void 0)
    return { serviceQuotaExceededException: obj.serviceQuotaExceededException };
  if (obj.throttlingException !== void 0)
    return { throttlingException: obj.throttlingException };
  if (obj.accessDeniedException !== void 0)
    return { accessDeniedException: obj.accessDeniedException };
  if (obj.conflictException !== void 0)
    return { conflictException: obj.conflictException };
  if (obj.dependencyFailedException !== void 0)
    return { dependencyFailedException: obj.dependencyFailedException };
  if (obj.badGatewayException !== void 0)
    return { badGatewayException: obj.badGatewayException };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "FlowResponseStreamFilterSensitiveLog");
var InvokeFlowResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.responseStream && { responseStream: "STREAMING_CONTENT" }
}), "InvokeFlowResponseFilterSensitiveLog");
var ByteContentFileFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.data && { data: import_smithy_client.SENSITIVE_STRING }
}), "ByteContentFileFilterSensitiveLog");
var FileSourceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.byteContent && { byteContent: ByteContentFileFilterSensitiveLog(obj.byteContent) }
}), "FileSourceFilterSensitiveLog");
var InputFileFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.source && { source: FileSourceFilterSensitiveLog(obj.source) }
}), "InputFileFilterSensitiveLog");
var ApiResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.apiPath && { apiPath: import_smithy_client.SENSITIVE_STRING }
}), "ApiResultFilterSensitiveLog");
var InvocationResultMemberFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.apiResult !== void 0)
    return { apiResult: ApiResultFilterSensitiveLog(obj.apiResult) };
  if (obj.functionResult !== void 0)
    return { functionResult: obj.functionResult };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "InvocationResultMemberFilterSensitiveLog");
var TextResponsePartFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "TextResponsePartFilterSensitiveLog");
var GeneratedResponsePartFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.textResponsePart && { textResponsePart: import_smithy_client.SENSITIVE_STRING }
}), "GeneratedResponsePartFilterSensitiveLog");
var RetrievalResultContentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "RetrievalResultContentFilterSensitiveLog");
var RetrievalResultLocationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "RetrievalResultLocationFilterSensitiveLog");
var RetrievedReferenceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.content && { content: import_smithy_client.SENSITIVE_STRING },
  ...obj.location && { location: import_smithy_client.SENSITIVE_STRING },
  ...obj.metadata && { metadata: import_smithy_client.SENSITIVE_STRING }
}), "RetrievedReferenceFilterSensitiveLog");
var CitationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.generatedResponsePart && {
    generatedResponsePart: GeneratedResponsePartFilterSensitiveLog(obj.generatedResponsePart)
  },
  ...obj.retrievedReferences && {
    retrievedReferences: obj.retrievedReferences.map((item) => RetrievedReferenceFilterSensitiveLog(item))
  }
}), "CitationFilterSensitiveLog");
var AttributionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.citations && { citations: obj.citations.map((item) => CitationFilterSensitiveLog(item)) }
}), "AttributionFilterSensitiveLog");
var PayloadPartFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.bytes && { bytes: import_smithy_client.SENSITIVE_STRING },
  ...obj.attribution && { attribution: AttributionFilterSensitiveLog(obj.attribution) }
}), "PayloadPartFilterSensitiveLog");
var OutputFileFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.bytes && { bytes: import_smithy_client.SENSITIVE_STRING }
}), "OutputFileFilterSensitiveLog");
var FilePartFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.files && { files: import_smithy_client.SENSITIVE_STRING }
}), "FilePartFilterSensitiveLog");
var ApiInvocationInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.apiPath && { apiPath: import_smithy_client.SENSITIVE_STRING }
}), "ApiInvocationInputFilterSensitiveLog");
var InvocationInputMemberFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.apiInvocationInput !== void 0)
    return { apiInvocationInput: ApiInvocationInputFilterSensitiveLog(obj.apiInvocationInput) };
  if (obj.functionInvocationInput !== void 0)
    return { functionInvocationInput: obj.functionInvocationInput };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "InvocationInputMemberFilterSensitiveLog");
var ReturnControlPayloadFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.invocationInputs && {
    invocationInputs: obj.invocationInputs.map((item) => InvocationInputMemberFilterSensitiveLog(item))
  }
}), "ReturnControlPayloadFilterSensitiveLog");
var FailureTraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.failureReason && { failureReason: import_smithy_client.SENSITIVE_STRING }
}), "FailureTraceFilterSensitiveLog");
var GuardrailContentFilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailContentFilterFilterSensitiveLog");
var GuardrailContentPolicyAssessmentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.filters && { filters: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailContentPolicyAssessmentFilterSensitiveLog");
var GuardrailPiiEntityFilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailPiiEntityFilterFilterSensitiveLog");
var GuardrailRegexFilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailRegexFilterFilterSensitiveLog");
var GuardrailSensitiveInformationPolicyAssessmentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.piiEntities && { piiEntities: import_smithy_client.SENSITIVE_STRING },
  ...obj.regexes && { regexes: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailSensitiveInformationPolicyAssessmentFilterSensitiveLog");
var GuardrailTopicFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailTopicFilterSensitiveLog");
var GuardrailTopicPolicyAssessmentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.topics && { topics: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailTopicPolicyAssessmentFilterSensitiveLog");
var GuardrailCustomWordFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailCustomWordFilterSensitiveLog");
var GuardrailManagedWordFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "GuardrailManagedWordFilterSensitiveLog");
var GuardrailWordPolicyAssessmentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.customWords && { customWords: import_smithy_client.SENSITIVE_STRING },
  ...obj.managedWordLists && { managedWordLists: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailWordPolicyAssessmentFilterSensitiveLog");
var GuardrailAssessmentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.topicPolicy && { topicPolicy: import_smithy_client.SENSITIVE_STRING },
  ...obj.contentPolicy && { contentPolicy: import_smithy_client.SENSITIVE_STRING },
  ...obj.wordPolicy && { wordPolicy: import_smithy_client.SENSITIVE_STRING },
  ...obj.sensitiveInformationPolicy && { sensitiveInformationPolicy: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailAssessmentFilterSensitiveLog");
var GuardrailTraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.inputAssessments && { inputAssessments: import_smithy_client.SENSITIVE_STRING },
  ...obj.outputAssessments && { outputAssessments: import_smithy_client.SENSITIVE_STRING }
}), "GuardrailTraceFilterSensitiveLog");
var KnowledgeBaseLookupInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING },
  ...obj.knowledgeBaseId && { knowledgeBaseId: import_smithy_client.SENSITIVE_STRING }
}), "KnowledgeBaseLookupInputFilterSensitiveLog");
var InvocationInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.actionGroupInvocationInput && {
    actionGroupInvocationInput: ActionGroupInvocationInputFilterSensitiveLog(obj.actionGroupInvocationInput)
  },
  ...obj.knowledgeBaseLookupInput && {
    knowledgeBaseLookupInput: KnowledgeBaseLookupInputFilterSensitiveLog(obj.knowledgeBaseLookupInput)
  }
}), "InvocationInputFilterSensitiveLog");
var ModelInvocationInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING }
}), "ModelInvocationInputFilterSensitiveLog");
var UsageFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "UsageFilterSensitiveLog");
var MetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.usage && { usage: import_smithy_client.SENSITIVE_STRING }
}), "MetadataFilterSensitiveLog");
var RawResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "RawResponseFilterSensitiveLog");
var OrchestrationModelInvocationOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.rawResponse && { rawResponse: import_smithy_client.SENSITIVE_STRING },
  ...obj.metadata && { metadata: import_smithy_client.SENSITIVE_STRING }
}), "OrchestrationModelInvocationOutputFilterSensitiveLog");
var FinalResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING }
}), "FinalResponseFilterSensitiveLog");
var KnowledgeBaseLookupOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.retrievedReferences && {
    retrievedReferences: obj.retrievedReferences.map((item) => RetrievedReferenceFilterSensitiveLog(item))
  }
}), "KnowledgeBaseLookupOutputFilterSensitiveLog");
var RepromptResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.source && { source: import_smithy_client.SENSITIVE_STRING }
}), "RepromptResponseFilterSensitiveLog");
var ObservationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.actionGroupInvocationOutput && {
    actionGroupInvocationOutput: ActionGroupInvocationOutputFilterSensitiveLog(obj.actionGroupInvocationOutput)
  },
  ...obj.knowledgeBaseLookupOutput && {
    knowledgeBaseLookupOutput: KnowledgeBaseLookupOutputFilterSensitiveLog(obj.knowledgeBaseLookupOutput)
  },
  ...obj.finalResponse && { finalResponse: FinalResponseFilterSensitiveLog(obj.finalResponse) },
  ...obj.repromptResponse && { repromptResponse: import_smithy_client.SENSITIVE_STRING }
}), "ObservationFilterSensitiveLog");
var RationaleFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING }
}), "RationaleFilterSensitiveLog");
var OrchestrationTraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.rationale !== void 0)
    return { rationale: import_smithy_client.SENSITIVE_STRING };
  if (obj.invocationInput !== void 0)
    return { invocationInput: import_smithy_client.SENSITIVE_STRING };
  if (obj.observation !== void 0)
    return { observation: import_smithy_client.SENSITIVE_STRING };
  if (obj.modelInvocationInput !== void 0)
    return { modelInvocationInput: import_smithy_client.SENSITIVE_STRING };
  if (obj.modelInvocationOutput !== void 0)
    return { modelInvocationOutput: import_smithy_client.SENSITIVE_STRING };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "OrchestrationTraceFilterSensitiveLog");
var PostProcessingParsedResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.text && { text: import_smithy_client.SENSITIVE_STRING }
}), "PostProcessingParsedResponseFilterSensitiveLog");
var PostProcessingModelInvocationOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.parsedResponse && { parsedResponse: import_smithy_client.SENSITIVE_STRING },
  ...obj.rawResponse && { rawResponse: import_smithy_client.SENSITIVE_STRING },
  ...obj.metadata && { metadata: import_smithy_client.SENSITIVE_STRING }
}), "PostProcessingModelInvocationOutputFilterSensitiveLog");
var PostProcessingTraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.modelInvocationInput !== void 0)
    return { modelInvocationInput: import_smithy_client.SENSITIVE_STRING };
  if (obj.modelInvocationOutput !== void 0)
    return { modelInvocationOutput: import_smithy_client.SENSITIVE_STRING };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "PostProcessingTraceFilterSensitiveLog");
var PreProcessingParsedResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.rationale && { rationale: import_smithy_client.SENSITIVE_STRING }
}), "PreProcessingParsedResponseFilterSensitiveLog");
var PreProcessingModelInvocationOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.parsedResponse && { parsedResponse: import_smithy_client.SENSITIVE_STRING },
  ...obj.rawResponse && { rawResponse: import_smithy_client.SENSITIVE_STRING },
  ...obj.metadata && { metadata: import_smithy_client.SENSITIVE_STRING }
}), "PreProcessingModelInvocationOutputFilterSensitiveLog");
var PreProcessingTraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.modelInvocationInput !== void 0)
    return { modelInvocationInput: import_smithy_client.SENSITIVE_STRING };
  if (obj.modelInvocationOutput !== void 0)
    return { modelInvocationOutput: import_smithy_client.SENSITIVE_STRING };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "PreProcessingTraceFilterSensitiveLog");
var TraceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.guardrailTrace !== void 0)
    return { guardrailTrace: import_smithy_client.SENSITIVE_STRING };
  if (obj.preProcessingTrace !== void 0)
    return { preProcessingTrace: import_smithy_client.SENSITIVE_STRING };
  if (obj.orchestrationTrace !== void 0)
    return { orchestrationTrace: import_smithy_client.SENSITIVE_STRING };
  if (obj.postProcessingTrace !== void 0)
    return { postProcessingTrace: import_smithy_client.SENSITIVE_STRING };
  if (obj.failureTrace !== void 0)
    return { failureTrace: import_smithy_client.SENSITIVE_STRING };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "TraceFilterSensitiveLog");
var TracePartFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.trace && { trace: import_smithy_client.SENSITIVE_STRING }
}), "TracePartFilterSensitiveLog");
var ResponseStreamFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.chunk !== void 0)
    return { chunk: import_smithy_client.SENSITIVE_STRING };
  if (obj.trace !== void 0)
    return { trace: import_smithy_client.SENSITIVE_STRING };
  if (obj.returnControl !== void 0)
    return { returnControl: import_smithy_client.SENSITIVE_STRING };
  if (obj.internalServerException !== void 0)
    return { internalServerException: obj.internalServerException };
  if (obj.validationException !== void 0)
    return { validationException: obj.validationException };
  if (obj.resourceNotFoundException !== void 0)
    return { resourceNotFoundException: obj.resourceNotFoundException };
  if (obj.serviceQuotaExceededException !== void 0)
    return { serviceQuotaExceededException: obj.serviceQuotaExceededException };
  if (obj.throttlingException !== void 0)
    return { throttlingException: obj.throttlingException };
  if (obj.accessDeniedException !== void 0)
    return { accessDeniedException: obj.accessDeniedException };
  if (obj.conflictException !== void 0)
    return { conflictException: obj.conflictException };
  if (obj.dependencyFailedException !== void 0)
    return { dependencyFailedException: obj.dependencyFailedException };
  if (obj.badGatewayException !== void 0)
    return { badGatewayException: obj.badGatewayException };
  if (obj.files !== void 0)
    return { files: FilePartFilterSensitiveLog(obj.files) };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "ResponseStreamFilterSensitiveLog");
var InvokeAgentResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.completion && { completion: "STREAMING_CONTENT" }
}), "InvokeAgentResponseFilterSensitiveLog");
var RetrieveAndGenerateInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "RetrieveAndGenerateInputFilterSensitiveLog");
var PromptTemplateFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.textPromptTemplate && { textPromptTemplate: import_smithy_client.SENSITIVE_STRING }
}), "PromptTemplateFilterSensitiveLog");
var ExternalSourcesGenerationConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.promptTemplate && { promptTemplate: PromptTemplateFilterSensitiveLog(obj.promptTemplate) }
}), "ExternalSourcesGenerationConfigurationFilterSensitiveLog");
var ByteContentDocFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identifier && { identifier: import_smithy_client.SENSITIVE_STRING },
  ...obj.data && { data: import_smithy_client.SENSITIVE_STRING }
}), "ByteContentDocFilterSensitiveLog");
var ExternalSourceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.byteContent && { byteContent: ByteContentDocFilterSensitiveLog(obj.byteContent) }
}), "ExternalSourceFilterSensitiveLog");
var ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.sources && { sources: obj.sources.map((item) => ExternalSourceFilterSensitiveLog(item)) },
  ...obj.generationConfiguration && {
    generationConfiguration: ExternalSourcesGenerationConfigurationFilterSensitiveLog(obj.generationConfiguration)
  }
}), "ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog");
var GenerationConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.promptTemplate && { promptTemplate: PromptTemplateFilterSensitiveLog(obj.promptTemplate) }
}), "GenerationConfigurationFilterSensitiveLog");
var OrchestrationConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.promptTemplate && { promptTemplate: PromptTemplateFilterSensitiveLog(obj.promptTemplate) }
}), "OrchestrationConfigurationFilterSensitiveLog");
var RetrieveAndGenerateOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "RetrieveAndGenerateOutputFilterSensitiveLog");
var RetrieveAndGenerateResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.output && { output: import_smithy_client.SENSITIVE_STRING },
  ...obj.citations && { citations: obj.citations.map((item) => CitationFilterSensitiveLog(item)) }
}), "RetrieveAndGenerateResponseFilterSensitiveLog");
var KnowledgeBaseQueryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "KnowledgeBaseQueryFilterSensitiveLog");
var KnowledgeBaseRetrievalResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.content && { content: import_smithy_client.SENSITIVE_STRING },
  ...obj.location && { location: import_smithy_client.SENSITIVE_STRING },
  ...obj.metadata && { metadata: import_smithy_client.SENSITIVE_STRING }
}), "KnowledgeBaseRetrievalResultFilterSensitiveLog");
var RetrieveResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.retrievalResults && { retrievalResults: import_smithy_client.SENSITIVE_STRING }
}), "RetrieveResponseFilterSensitiveLog");
var RetrievalFilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => {
  if (obj.equals !== void 0)
    return { equals: obj.equals };
  if (obj.notEquals !== void 0)
    return { notEquals: obj.notEquals };
  if (obj.greaterThan !== void 0)
    return { greaterThan: obj.greaterThan };
  if (obj.greaterThanOrEquals !== void 0)
    return { greaterThanOrEquals: obj.greaterThanOrEquals };
  if (obj.lessThan !== void 0)
    return { lessThan: obj.lessThan };
  if (obj.lessThanOrEquals !== void 0)
    return { lessThanOrEquals: obj.lessThanOrEquals };
  if (obj.in !== void 0)
    return { in: obj.in };
  if (obj.notIn !== void 0)
    return { notIn: obj.notIn };
  if (obj.startsWith !== void 0)
    return { startsWith: obj.startsWith };
  if (obj.listContains !== void 0)
    return { listContains: obj.listContains };
  if (obj.stringContains !== void 0)
    return { stringContains: obj.stringContains };
  if (obj.andAll !== void 0)
    return { andAll: import_smithy_client.SENSITIVE_STRING };
  if (obj.orAll !== void 0)
    return { orAll: import_smithy_client.SENSITIVE_STRING };
  if (obj.$unknown !== void 0)
    return { [obj.$unknown[0]]: "UNKNOWN" };
}, "RetrievalFilterFilterSensitiveLog");
var KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.filter && { filter: import_smithy_client.SENSITIVE_STRING }
}), "KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog");
var KnowledgeBaseRetrievalConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.vectorSearchConfiguration && {
    vectorSearchConfiguration: KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog(obj.vectorSearchConfiguration)
  }
}), "KnowledgeBaseRetrievalConfigurationFilterSensitiveLog");
var KnowledgeBaseConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.retrievalConfiguration && {
    retrievalConfiguration: KnowledgeBaseRetrievalConfigurationFilterSensitiveLog(obj.retrievalConfiguration)
  }
}), "KnowledgeBaseConfigurationFilterSensitiveLog");
var KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.retrievalConfiguration && {
    retrievalConfiguration: KnowledgeBaseRetrievalConfigurationFilterSensitiveLog(obj.retrievalConfiguration)
  },
  ...obj.generationConfiguration && {
    generationConfiguration: GenerationConfigurationFilterSensitiveLog(obj.generationConfiguration)
  },
  ...obj.orchestrationConfiguration && {
    orchestrationConfiguration: OrchestrationConfigurationFilterSensitiveLog(obj.orchestrationConfiguration)
  }
}), "KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog");
var RetrieveRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.retrievalQuery && { retrievalQuery: import_smithy_client.SENSITIVE_STRING },
  ...obj.retrievalConfiguration && {
    retrievalConfiguration: KnowledgeBaseRetrievalConfigurationFilterSensitiveLog(obj.retrievalConfiguration)
  }
}), "RetrieveRequestFilterSensitiveLog");
var RetrieveAndGenerateConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.knowledgeBaseConfiguration && {
    knowledgeBaseConfiguration: KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog(
      obj.knowledgeBaseConfiguration
    )
  },
  ...obj.externalSourcesConfiguration && {
    externalSourcesConfiguration: ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog(
      obj.externalSourcesConfiguration
    )
  }
}), "RetrieveAndGenerateConfigurationFilterSensitiveLog");
var RetrieveAndGenerateRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.input && { input: import_smithy_client.SENSITIVE_STRING },
  ...obj.retrieveAndGenerateConfiguration && {
    retrieveAndGenerateConfiguration: RetrieveAndGenerateConfigurationFilterSensitiveLog(
      obj.retrieveAndGenerateConfiguration
    )
  }
}), "RetrieveAndGenerateRequestFilterSensitiveLog");
var SessionStateFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.returnControlInvocationResults && {
    returnControlInvocationResults: obj.returnControlInvocationResults.map(
      (item) => InvocationResultMemberFilterSensitiveLog(item)
    )
  },
  ...obj.files && { files: obj.files.map((item) => InputFileFilterSensitiveLog(item)) },
  ...obj.knowledgeBaseConfigurations && {
    knowledgeBaseConfigurations: obj.knowledgeBaseConfigurations.map(
      (item) => KnowledgeBaseConfigurationFilterSensitiveLog(item)
    )
  }
}), "SessionStateFilterSensitiveLog");
var InvokeAgentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.sessionState && { sessionState: SessionStateFilterSensitiveLog(obj.sessionState) },
  ...obj.inputText && { inputText: import_smithy_client.SENSITIVE_STRING }
}), "InvokeAgentRequestFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_DeleteAgentMemoryCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/agents/{agentId}/agentAliases/{agentAliasId}/memories");
  b.p("agentId", () => input.agentId, "{agentId}", false);
  b.p("agentAliasId", () => input.agentAliasId, "{agentAliasId}", false);
  const query = (0, import_smithy_client.map)({
    [_mI]: [, input[_mI]]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_DeleteAgentMemoryCommand");
var se_GetAgentMemoryCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/agents/{agentId}/agentAliases/{agentAliasId}/memories");
  b.p("agentId", () => input.agentId, "{agentId}", false);
  b.p("agentAliasId", () => input.agentAliasId, "{agentAliasId}", false);
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mIa]: [() => input.maxItems !== void 0, () => input[_mIa].toString()],
    [_mT]: [, (0, import_smithy_client.expectNonNull)(input[_mT], `memoryType`)],
    [_mI]: [, (0, import_smithy_client.expectNonNull)(input[_mI], `memoryId`)]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetAgentMemoryCommand");
var se_InvokeAgentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/agents/{agentId}/agentAliases/{agentAliasId}/sessions/{sessionId}/text");
  b.p("agentId", () => input.agentId, "{agentId}", false);
  b.p("agentAliasId", () => input.agentAliasId, "{agentAliasId}", false);
  b.p("sessionId", () => input.sessionId, "{sessionId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      enableTrace: [],
      endSession: [],
      inputText: [],
      memoryId: [],
      sessionState: (_) => se_SessionState(_, context)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_InvokeAgentCommand");
var se_InvokeFlowCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}");
  b.p("flowIdentifier", () => input.flowIdentifier, "{flowIdentifier}", false);
  b.p("flowAliasIdentifier", () => input.flowAliasIdentifier, "{flowAliasIdentifier}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      inputs: (_) => se_FlowInputs(_, context)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_InvokeFlowCommand");
var se_RetrieveCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/knowledgebases/{knowledgeBaseId}/retrieve");
  b.p("knowledgeBaseId", () => input.knowledgeBaseId, "{knowledgeBaseId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      nextToken: [],
      retrievalConfiguration: (_) => se_KnowledgeBaseRetrievalConfiguration(_, context),
      retrievalQuery: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_RetrieveCommand");
var se_RetrieveAndGenerateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/retrieveAndGenerate");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      input: (_) => (0, import_smithy_client._json)(_),
      retrieveAndGenerateConfiguration: (_) => se_RetrieveAndGenerateConfiguration(_, context),
      sessionConfiguration: (_) => (0, import_smithy_client._json)(_),
      sessionId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_RetrieveAndGenerateCommand");
var de_DeleteAgentMemoryCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteAgentMemoryCommand");
var de_GetAgentMemoryCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    memoryContents: (_) => de_Memories(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetAgentMemoryCommand");
var de_InvokeAgentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output),
    [_cT]: [, output.headers[_xabact]],
    [_sI]: [, output.headers[_xabasi]],
    [_mI]: [, output.headers[_xabami]]
  });
  const data = output.body;
  contents.completion = de_ResponseStream(data, context);
  return contents;
}, "de_InvokeAgentCommand");
var de_InvokeFlowCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = output.body;
  contents.responseStream = de_FlowResponseStream(data, context);
  return contents;
}, "de_InvokeFlowCommand");
var de_RetrieveCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    retrievalResults: (_) => de_KnowledgeBaseRetrievalResults(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_RetrieveCommand");
var de_RetrieveAndGenerateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    citations: (_) => de_Citations(_, context),
    guardrailAction: import_smithy_client.expectString,
    output: import_smithy_client._json,
    sessionId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_RetrieveAndGenerateCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.bedrockagentruntime#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "BadGatewayException":
    case "com.amazonaws.bedrockagentruntime#BadGatewayException":
      throw await de_BadGatewayExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.bedrockagentruntime#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "DependencyFailedException":
    case "com.amazonaws.bedrockagentruntime#DependencyFailedException":
      throw await de_DependencyFailedExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.bedrockagentruntime#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.bedrockagentruntime#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.bedrockagentruntime#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.bedrockagentruntime#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.bedrockagentruntime#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(BedrockAgentRuntimeServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_BadGatewayExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new BadGatewayException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_BadGatewayExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_DependencyFailedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DependencyFailedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DependencyFailedExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var de_FlowResponseStream = /* @__PURE__ */ __name((output, context) => {
  return context.eventStreamMarshaller.deserialize(output, async (event) => {
    if (event["flowOutputEvent"] != null) {
      return {
        flowOutputEvent: await de_FlowOutputEvent_event(event["flowOutputEvent"], context)
      };
    }
    if (event["flowCompletionEvent"] != null) {
      return {
        flowCompletionEvent: await de_FlowCompletionEvent_event(event["flowCompletionEvent"], context)
      };
    }
    if (event["internalServerException"] != null) {
      return {
        internalServerException: await de_InternalServerException_event(event["internalServerException"], context)
      };
    }
    if (event["validationException"] != null) {
      return {
        validationException: await de_ValidationException_event(event["validationException"], context)
      };
    }
    if (event["resourceNotFoundException"] != null) {
      return {
        resourceNotFoundException: await de_ResourceNotFoundException_event(
          event["resourceNotFoundException"],
          context
        )
      };
    }
    if (event["serviceQuotaExceededException"] != null) {
      return {
        serviceQuotaExceededException: await de_ServiceQuotaExceededException_event(
          event["serviceQuotaExceededException"],
          context
        )
      };
    }
    if (event["throttlingException"] != null) {
      return {
        throttlingException: await de_ThrottlingException_event(event["throttlingException"], context)
      };
    }
    if (event["accessDeniedException"] != null) {
      return {
        accessDeniedException: await de_AccessDeniedException_event(event["accessDeniedException"], context)
      };
    }
    if (event["conflictException"] != null) {
      return {
        conflictException: await de_ConflictException_event(event["conflictException"], context)
      };
    }
    if (event["dependencyFailedException"] != null) {
      return {
        dependencyFailedException: await de_DependencyFailedException_event(
          event["dependencyFailedException"],
          context
        )
      };
    }
    if (event["badGatewayException"] != null) {
      return {
        badGatewayException: await de_BadGatewayException_event(event["badGatewayException"], context)
      };
    }
    return { $unknown: output };
  });
}, "de_FlowResponseStream");
var de_ResponseStream = /* @__PURE__ */ __name((output, context) => {
  return context.eventStreamMarshaller.deserialize(output, async (event) => {
    if (event["chunk"] != null) {
      return {
        chunk: await de_PayloadPart_event(event["chunk"], context)
      };
    }
    if (event["trace"] != null) {
      return {
        trace: await de_TracePart_event(event["trace"], context)
      };
    }
    if (event["returnControl"] != null) {
      return {
        returnControl: await de_ReturnControlPayload_event(event["returnControl"], context)
      };
    }
    if (event["internalServerException"] != null) {
      return {
        internalServerException: await de_InternalServerException_event(event["internalServerException"], context)
      };
    }
    if (event["validationException"] != null) {
      return {
        validationException: await de_ValidationException_event(event["validationException"], context)
      };
    }
    if (event["resourceNotFoundException"] != null) {
      return {
        resourceNotFoundException: await de_ResourceNotFoundException_event(
          event["resourceNotFoundException"],
          context
        )
      };
    }
    if (event["serviceQuotaExceededException"] != null) {
      return {
        serviceQuotaExceededException: await de_ServiceQuotaExceededException_event(
          event["serviceQuotaExceededException"],
          context
        )
      };
    }
    if (event["throttlingException"] != null) {
      return {
        throttlingException: await de_ThrottlingException_event(event["throttlingException"], context)
      };
    }
    if (event["accessDeniedException"] != null) {
      return {
        accessDeniedException: await de_AccessDeniedException_event(event["accessDeniedException"], context)
      };
    }
    if (event["conflictException"] != null) {
      return {
        conflictException: await de_ConflictException_event(event["conflictException"], context)
      };
    }
    if (event["dependencyFailedException"] != null) {
      return {
        dependencyFailedException: await de_DependencyFailedException_event(
          event["dependencyFailedException"],
          context
        )
      };
    }
    if (event["badGatewayException"] != null) {
      return {
        badGatewayException: await de_BadGatewayException_event(event["badGatewayException"], context)
      };
    }
    if (event["files"] != null) {
      return {
        files: await de_FilePart_event(event["files"], context)
      };
    }
    return { $unknown: output };
  });
}, "de_ResponseStream");
var de_AccessDeniedException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_AccessDeniedExceptionRes(parsedOutput, context);
}, "de_AccessDeniedException_event");
var de_BadGatewayException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_BadGatewayExceptionRes(parsedOutput, context);
}, "de_BadGatewayException_event");
var de_ConflictException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_ConflictExceptionRes(parsedOutput, context);
}, "de_ConflictException_event");
var de_DependencyFailedException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_DependencyFailedExceptionRes(parsedOutput, context);
}, "de_DependencyFailedException_event");
var de_FilePart_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, de_FilePart(data, context));
  return contents;
}, "de_FilePart_event");
var de_FlowCompletionEvent_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, (0, import_smithy_client._json)(data));
  return contents;
}, "de_FlowCompletionEvent_event");
var de_FlowOutputEvent_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, de_FlowOutputEvent(data, context));
  return contents;
}, "de_FlowOutputEvent_event");
var de_InternalServerException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_InternalServerExceptionRes(parsedOutput, context);
}, "de_InternalServerException_event");
var de_PayloadPart_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, de_PayloadPart(data, context));
  return contents;
}, "de_PayloadPart_event");
var de_ResourceNotFoundException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_ResourceNotFoundExceptionRes(parsedOutput, context);
}, "de_ResourceNotFoundException_event");
var de_ReturnControlPayload_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, (0, import_smithy_client._json)(data));
  return contents;
}, "de_ReturnControlPayload_event");
var de_ServiceQuotaExceededException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
}, "de_ServiceQuotaExceededException_event");
var de_ThrottlingException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_ThrottlingExceptionRes(parsedOutput, context);
}, "de_ThrottlingException_event");
var de_TracePart_event = /* @__PURE__ */ __name(async (output, context) => {
  const contents = {};
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  Object.assign(contents, de_TracePart(data, context));
  return contents;
}, "de_TracePart_event");
var de_ValidationException_event = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonBody)(output.body, context)
  };
  return de_ValidationExceptionRes(parsedOutput, context);
}, "de_ValidationException_event");
var se_AdditionalModelRequestFields = /* @__PURE__ */ __name((input, context) => {
  return Object.entries(input).reduce((acc, [key, value]) => {
    if (value === null) {
      return acc;
    }
    acc[key] = se_AdditionalModelRequestFieldsValue(value, context);
    return acc;
  }, {});
}, "se_AdditionalModelRequestFields");
var se_AdditionalModelRequestFieldsValue = /* @__PURE__ */ __name((input, context) => {
  return input;
}, "se_AdditionalModelRequestFieldsValue");
var se_ByteContentDoc = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    contentType: [],
    data: context.base64Encoder,
    identifier: []
  });
}, "se_ByteContentDoc");
var se_ByteContentFile = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    data: context.base64Encoder,
    mediaType: []
  });
}, "se_ByteContentFile");
var se_ExternalSource = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    byteContent: (_) => se_ByteContentDoc(_, context),
    s3Location: import_smithy_client._json,
    sourceType: []
  });
}, "se_ExternalSource");
var se_ExternalSources = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_ExternalSource(entry, context);
  });
}, "se_ExternalSources");
var se_ExternalSourcesGenerationConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    additionalModelRequestFields: (_) => se_AdditionalModelRequestFields(_, context),
    guardrailConfiguration: import_smithy_client._json,
    inferenceConfig: (_) => se_InferenceConfig(_, context),
    promptTemplate: import_smithy_client._json
  });
}, "se_ExternalSourcesGenerationConfiguration");
var se_ExternalSourcesRetrieveAndGenerateConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    generationConfiguration: (_) => se_ExternalSourcesGenerationConfiguration(_, context),
    modelArn: [],
    sources: (_) => se_ExternalSources(_, context)
  });
}, "se_ExternalSourcesRetrieveAndGenerateConfiguration");
var se_FileSource = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    byteContent: (_) => se_ByteContentFile(_, context),
    s3Location: import_smithy_client._json,
    sourceType: []
  });
}, "se_FileSource");
var se_FilterAttribute = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    key: [],
    value: (_) => se_FilterValue(_, context)
  });
}, "se_FilterAttribute");
var se_FilterValue = /* @__PURE__ */ __name((input, context) => {
  return input;
}, "se_FilterValue");
var se_FlowInput = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    content: (_) => se_FlowInputContent(_, context),
    nodeName: [],
    nodeOutputName: []
  });
}, "se_FlowInput");
var se_FlowInputContent = /* @__PURE__ */ __name((input, context) => {
  return FlowInputContent.visit(input, {
    document: (value) => ({ document: se_Document(value, context) }),
    _: (name, value) => ({ name: value })
  });
}, "se_FlowInputContent");
var se_FlowInputs = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_FlowInput(entry, context);
  });
}, "se_FlowInputs");
var se_GenerationConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    additionalModelRequestFields: (_) => se_AdditionalModelRequestFields(_, context),
    guardrailConfiguration: import_smithy_client._json,
    inferenceConfig: (_) => se_InferenceConfig(_, context),
    promptTemplate: import_smithy_client._json
  });
}, "se_GenerationConfiguration");
var se_InferenceConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    textInferenceConfig: (_) => se_TextInferenceConfig(_, context)
  });
}, "se_InferenceConfig");
var se_InputFile = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    name: [],
    source: (_) => se_FileSource(_, context),
    useCase: []
  });
}, "se_InputFile");
var se_InputFiles = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_InputFile(entry, context);
  });
}, "se_InputFiles");
var se_KnowledgeBaseConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    knowledgeBaseId: [],
    retrievalConfiguration: (_) => se_KnowledgeBaseRetrievalConfiguration(_, context)
  });
}, "se_KnowledgeBaseConfiguration");
var se_KnowledgeBaseConfigurations = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_KnowledgeBaseConfiguration(entry, context);
  });
}, "se_KnowledgeBaseConfigurations");
var se_KnowledgeBaseRetrievalConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    vectorSearchConfiguration: (_) => se_KnowledgeBaseVectorSearchConfiguration(_, context)
  });
}, "se_KnowledgeBaseRetrievalConfiguration");
var se_KnowledgeBaseRetrieveAndGenerateConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    generationConfiguration: (_) => se_GenerationConfiguration(_, context),
    knowledgeBaseId: [],
    modelArn: [],
    orchestrationConfiguration: (_) => se_OrchestrationConfiguration(_, context),
    retrievalConfiguration: (_) => se_KnowledgeBaseRetrievalConfiguration(_, context)
  });
}, "se_KnowledgeBaseRetrieveAndGenerateConfiguration");
var se_KnowledgeBaseVectorSearchConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    filter: (_) => se_RetrievalFilter(_, context),
    numberOfResults: [],
    overrideSearchType: []
  });
}, "se_KnowledgeBaseVectorSearchConfiguration");
var se_OrchestrationConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    additionalModelRequestFields: (_) => se_AdditionalModelRequestFields(_, context),
    inferenceConfig: (_) => se_InferenceConfig(_, context),
    promptTemplate: import_smithy_client._json,
    queryTransformationConfiguration: import_smithy_client._json
  });
}, "se_OrchestrationConfiguration");
var se_RetrievalFilter = /* @__PURE__ */ __name((input, context) => {
  return RetrievalFilter.visit(input, {
    andAll: (value) => ({ andAll: se_RetrievalFilterList(value, context) }),
    equals: (value) => ({ equals: se_FilterAttribute(value, context) }),
    greaterThan: (value) => ({ greaterThan: se_FilterAttribute(value, context) }),
    greaterThanOrEquals: (value) => ({ greaterThanOrEquals: se_FilterAttribute(value, context) }),
    in: (value) => ({ in: se_FilterAttribute(value, context) }),
    lessThan: (value) => ({ lessThan: se_FilterAttribute(value, context) }),
    lessThanOrEquals: (value) => ({ lessThanOrEquals: se_FilterAttribute(value, context) }),
    listContains: (value) => ({ listContains: se_FilterAttribute(value, context) }),
    notEquals: (value) => ({ notEquals: se_FilterAttribute(value, context) }),
    notIn: (value) => ({ notIn: se_FilterAttribute(value, context) }),
    orAll: (value) => ({ orAll: se_RetrievalFilterList(value, context) }),
    startsWith: (value) => ({ startsWith: se_FilterAttribute(value, context) }),
    stringContains: (value) => ({ stringContains: se_FilterAttribute(value, context) }),
    _: (name, value) => ({ name: value })
  });
}, "se_RetrievalFilter");
var se_RetrievalFilterList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_RetrievalFilter(entry, context);
  });
}, "se_RetrievalFilterList");
var se_RetrieveAndGenerateConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    externalSourcesConfiguration: (_) => se_ExternalSourcesRetrieveAndGenerateConfiguration(_, context),
    knowledgeBaseConfiguration: (_) => se_KnowledgeBaseRetrieveAndGenerateConfiguration(_, context),
    type: []
  });
}, "se_RetrieveAndGenerateConfiguration");
var se_SessionState = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    files: (_) => se_InputFiles(_, context),
    invocationId: [],
    knowledgeBaseConfigurations: (_) => se_KnowledgeBaseConfigurations(_, context),
    promptSessionAttributes: import_smithy_client._json,
    returnControlInvocationResults: import_smithy_client._json,
    sessionAttributes: import_smithy_client._json
  });
}, "se_SessionState");
var se_TextInferenceConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    maxTokens: [],
    stopSequences: import_smithy_client._json,
    temperature: import_smithy_client.serializeFloat,
    topP: import_smithy_client.serializeFloat
  });
}, "se_TextInferenceConfig");
var se_Document = /* @__PURE__ */ __name((input, context) => {
  return input;
}, "se_Document");
var de_Attribution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    citations: (_) => de_Citations(_, context)
  });
}, "de_Attribution");
var de_Citation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    generatedResponsePart: import_smithy_client._json,
    retrievedReferences: (_) => de_RetrievedReferences(_, context)
  });
}, "de_Citation");
var de_Citations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Citation(entry, context);
  });
  return retVal;
}, "de_Citations");
var de_FilePart = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    files: (_) => de_OutputFiles(_, context)
  });
}, "de_FilePart");
var de_FlowOutputContent = /* @__PURE__ */ __name((output, context) => {
  if (output.document != null) {
    return {
      document: de_Document(output.document, context)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_FlowOutputContent");
var de_FlowOutputEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    content: (_) => de_FlowOutputContent((0, import_core2.awsExpectUnion)(_), context),
    nodeName: import_smithy_client.expectString,
    nodeType: import_smithy_client.expectString
  });
}, "de_FlowOutputEvent");
var de_InferenceConfiguration = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    maximumLength: import_smithy_client.expectInt32,
    stopSequences: import_smithy_client._json,
    temperature: import_smithy_client.limitedParseFloat32,
    topK: import_smithy_client.expectInt32,
    topP: import_smithy_client.limitedParseFloat32
  });
}, "de_InferenceConfiguration");
var de_KnowledgeBaseLookupOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    retrievedReferences: (_) => de_RetrievedReferences(_, context)
  });
}, "de_KnowledgeBaseLookupOutput");
var de_KnowledgeBaseRetrievalResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    content: import_smithy_client._json,
    location: import_smithy_client._json,
    metadata: (_) => de_RetrievalResultMetadata(_, context),
    score: import_smithy_client.limitedParseDouble
  });
}, "de_KnowledgeBaseRetrievalResult");
var de_KnowledgeBaseRetrievalResults = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_KnowledgeBaseRetrievalResult(entry, context);
  });
  return retVal;
}, "de_KnowledgeBaseRetrievalResults");
var de_Memories = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Memory((0, import_core2.awsExpectUnion)(entry), context);
  });
  return retVal;
}, "de_Memories");
var de_Memory = /* @__PURE__ */ __name((output, context) => {
  if (output.sessionSummary != null) {
    return {
      sessionSummary: de_MemorySessionSummary(output.sessionSummary, context)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_Memory");
var de_MemorySessionSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    memoryId: import_smithy_client.expectString,
    sessionExpiryTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    sessionId: import_smithy_client.expectString,
    sessionStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    summaryText: import_smithy_client.expectString
  });
}, "de_MemorySessionSummary");
var de_ModelInvocationInput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    inferenceConfiguration: (_) => de_InferenceConfiguration(_, context),
    overrideLambda: import_smithy_client.expectString,
    parserMode: import_smithy_client.expectString,
    promptCreationMode: import_smithy_client.expectString,
    text: import_smithy_client.expectString,
    traceId: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_ModelInvocationInput");
var de_Observation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionGroupInvocationOutput: import_smithy_client._json,
    codeInterpreterInvocationOutput: import_smithy_client._json,
    finalResponse: import_smithy_client._json,
    knowledgeBaseLookupOutput: (_) => de_KnowledgeBaseLookupOutput(_, context),
    repromptResponse: import_smithy_client._json,
    traceId: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_Observation");
var de_OrchestrationTrace = /* @__PURE__ */ __name((output, context) => {
  if (output.invocationInput != null) {
    return {
      invocationInput: (0, import_smithy_client._json)(output.invocationInput)
    };
  }
  if (output.modelInvocationInput != null) {
    return {
      modelInvocationInput: de_ModelInvocationInput(output.modelInvocationInput, context)
    };
  }
  if (output.modelInvocationOutput != null) {
    return {
      modelInvocationOutput: (0, import_smithy_client._json)(output.modelInvocationOutput)
    };
  }
  if (output.observation != null) {
    return {
      observation: de_Observation(output.observation, context)
    };
  }
  if (output.rationale != null) {
    return {
      rationale: (0, import_smithy_client._json)(output.rationale)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_OrchestrationTrace");
var de_OutputFile = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    bytes: context.base64Decoder,
    name: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_OutputFile");
var de_OutputFiles = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_OutputFile(entry, context);
  });
  return retVal;
}, "de_OutputFiles");
var de_PayloadPart = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    attribution: (_) => de_Attribution(_, context),
    bytes: context.base64Decoder
  });
}, "de_PayloadPart");
var de_PostProcessingTrace = /* @__PURE__ */ __name((output, context) => {
  if (output.modelInvocationInput != null) {
    return {
      modelInvocationInput: de_ModelInvocationInput(output.modelInvocationInput, context)
    };
  }
  if (output.modelInvocationOutput != null) {
    return {
      modelInvocationOutput: (0, import_smithy_client._json)(output.modelInvocationOutput)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_PostProcessingTrace");
var de_PreProcessingTrace = /* @__PURE__ */ __name((output, context) => {
  if (output.modelInvocationInput != null) {
    return {
      modelInvocationInput: de_ModelInvocationInput(output.modelInvocationInput, context)
    };
  }
  if (output.modelInvocationOutput != null) {
    return {
      modelInvocationOutput: (0, import_smithy_client._json)(output.modelInvocationOutput)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_PreProcessingTrace");
var de_RetrievalResultMetadata = /* @__PURE__ */ __name((output, context) => {
  return Object.entries(output).reduce((acc, [key, value]) => {
    if (value === null) {
      return acc;
    }
    acc[key] = de_RetrievalResultMetadataValue(value, context);
    return acc;
  }, {});
}, "de_RetrievalResultMetadata");
var de_RetrievalResultMetadataValue = /* @__PURE__ */ __name((output, context) => {
  return output;
}, "de_RetrievalResultMetadataValue");
var de_RetrievedReference = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    content: import_smithy_client._json,
    location: import_smithy_client._json,
    metadata: (_) => de_RetrievalResultMetadata(_, context)
  });
}, "de_RetrievedReference");
var de_RetrievedReferences = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RetrievedReference(entry, context);
  });
  return retVal;
}, "de_RetrievedReferences");
var de_Trace = /* @__PURE__ */ __name((output, context) => {
  if (output.failureTrace != null) {
    return {
      failureTrace: (0, import_smithy_client._json)(output.failureTrace)
    };
  }
  if (output.guardrailTrace != null) {
    return {
      guardrailTrace: (0, import_smithy_client._json)(output.guardrailTrace)
    };
  }
  if (output.orchestrationTrace != null) {
    return {
      orchestrationTrace: de_OrchestrationTrace((0, import_core2.awsExpectUnion)(output.orchestrationTrace), context)
    };
  }
  if (output.postProcessingTrace != null) {
    return {
      postProcessingTrace: de_PostProcessingTrace((0, import_core2.awsExpectUnion)(output.postProcessingTrace), context)
    };
  }
  if (output.preProcessingTrace != null) {
    return {
      preProcessingTrace: de_PreProcessingTrace((0, import_core2.awsExpectUnion)(output.preProcessingTrace), context)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_Trace");
var de_TracePart = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    agentAliasId: import_smithy_client.expectString,
    agentId: import_smithy_client.expectString,
    agentVersion: import_smithy_client.expectString,
    sessionId: import_smithy_client.expectString,
    trace: (_) => de_Trace((0, import_core2.awsExpectUnion)(_), context)
  });
}, "de_TracePart");
var de_Document = /* @__PURE__ */ __name((output, context) => {
  return output;
}, "de_Document");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _cT = "contentType";
var _mI = "memoryId";
var _mIa = "maxItems";
var _mT = "memoryType";
var _nT = "nextToken";
var _sI = "sessionId";
var _xabact = "x-amzn-bedrock-agent-content-type";
var _xabami = "x-amz-bedrock-agent-memory-id";
var _xabasi = "x-amz-bedrock-agent-session-id";

// src/commands/DeleteAgentMemoryCommand.ts
var _DeleteAgentMemoryCommand = class _DeleteAgentMemoryCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "DeleteAgentMemory", {}).n("BedrockAgentRuntimeClient", "DeleteAgentMemoryCommand").f(void 0, void 0).ser(se_DeleteAgentMemoryCommand).de(de_DeleteAgentMemoryCommand).build() {
};
__name(_DeleteAgentMemoryCommand, "DeleteAgentMemoryCommand");
var DeleteAgentMemoryCommand = _DeleteAgentMemoryCommand;

// src/commands/GetAgentMemoryCommand.ts



var _GetAgentMemoryCommand = class _GetAgentMemoryCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "GetAgentMemory", {}).n("BedrockAgentRuntimeClient", "GetAgentMemoryCommand").f(void 0, void 0).ser(se_GetAgentMemoryCommand).de(de_GetAgentMemoryCommand).build() {
};
__name(_GetAgentMemoryCommand, "GetAgentMemoryCommand");
var GetAgentMemoryCommand = _GetAgentMemoryCommand;

// src/commands/InvokeAgentCommand.ts



var _InvokeAgentCommand = class _InvokeAgentCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "InvokeAgent", {
  /**
   * @internal
   */
  eventStream: {
    output: true
  }
}).n("BedrockAgentRuntimeClient", "InvokeAgentCommand").f(InvokeAgentRequestFilterSensitiveLog, InvokeAgentResponseFilterSensitiveLog).ser(se_InvokeAgentCommand).de(de_InvokeAgentCommand).build() {
};
__name(_InvokeAgentCommand, "InvokeAgentCommand");
var InvokeAgentCommand = _InvokeAgentCommand;

// src/commands/InvokeFlowCommand.ts



var _InvokeFlowCommand = class _InvokeFlowCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "InvokeFlow", {
  /**
   * @internal
   */
  eventStream: {
    output: true
  }
}).n("BedrockAgentRuntimeClient", "InvokeFlowCommand").f(InvokeFlowRequestFilterSensitiveLog, InvokeFlowResponseFilterSensitiveLog).ser(se_InvokeFlowCommand).de(de_InvokeFlowCommand).build() {
};
__name(_InvokeFlowCommand, "InvokeFlowCommand");
var InvokeFlowCommand = _InvokeFlowCommand;

// src/commands/RetrieveAndGenerateCommand.ts



var _RetrieveAndGenerateCommand = class _RetrieveAndGenerateCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "RetrieveAndGenerate", {}).n("BedrockAgentRuntimeClient", "RetrieveAndGenerateCommand").f(RetrieveAndGenerateRequestFilterSensitiveLog, RetrieveAndGenerateResponseFilterSensitiveLog).ser(se_RetrieveAndGenerateCommand).de(de_RetrieveAndGenerateCommand).build() {
};
__name(_RetrieveAndGenerateCommand, "RetrieveAndGenerateCommand");
var RetrieveAndGenerateCommand = _RetrieveAndGenerateCommand;

// src/commands/RetrieveCommand.ts



var _RetrieveCommand = class _RetrieveCommand extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonBedrockAgentRunTimeService", "Retrieve", {}).n("BedrockAgentRuntimeClient", "RetrieveCommand").f(RetrieveRequestFilterSensitiveLog, RetrieveResponseFilterSensitiveLog).ser(se_RetrieveCommand).de(de_RetrieveCommand).build() {
};
__name(_RetrieveCommand, "RetrieveCommand");
var RetrieveCommand = _RetrieveCommand;

// src/BedrockAgentRuntime.ts
var commands = {
  DeleteAgentMemoryCommand,
  GetAgentMemoryCommand,
  InvokeAgentCommand,
  InvokeFlowCommand,
  RetrieveCommand,
  RetrieveAndGenerateCommand
};
var _BedrockAgentRuntime = class _BedrockAgentRuntime extends BedrockAgentRuntimeClient {
};
__name(_BedrockAgentRuntime, "BedrockAgentRuntime");
var BedrockAgentRuntime = _BedrockAgentRuntime;
(0, import_smithy_client.createAggregatedClient)(commands, BedrockAgentRuntime);

// src/pagination/GetAgentMemoryPaginator.ts

var paginateGetAgentMemory = (0, import_core.createPaginator)(BedrockAgentRuntimeClient, GetAgentMemoryCommand, "nextToken", "nextToken", "maxItems");

// src/pagination/RetrievePaginator.ts

var paginateRetrieve = (0, import_core.createPaginator)(BedrockAgentRuntimeClient, RetrieveCommand, "nextToken", "nextToken", "");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  BedrockAgentRuntimeServiceException,
  __Client,
  BedrockAgentRuntimeClient,
  BedrockAgentRuntime,
  $Command,
  DeleteAgentMemoryCommand,
  GetAgentMemoryCommand,
  InvokeAgentCommand,
  InvokeFlowCommand,
  RetrieveAndGenerateCommand,
  RetrieveCommand,
  paginateGetAgentMemory,
  paginateRetrieve,
  AccessDeniedException,
  ExecutionType,
  ActionInvocationType,
  BadGatewayException,
  ConflictException,
  DependencyFailedException,
  InternalServerException,
  FlowInputContent,
  FlowCompletionReason,
  FlowOutputContent,
  NodeType,
  ResourceNotFoundException,
  ServiceQuotaExceededException,
  ThrottlingException,
  ValidationException,
  FlowResponseStream,
  FileSourceType,
  FileUseCase,
  SearchType,
  ConfirmationState,
  ResponseState,
  InvocationResultMember,
  RetrievalResultLocationType,
  InvocationInputMember,
  GuardrailAction,
  GuardrailContentPolicyAction,
  GuardrailContentFilterConfidence,
  GuardrailContentFilterType,
  GuardrailSensitiveInformationPolicyAction,
  GuardrailPiiEntityType,
  GuardrailTopicPolicyAction,
  GuardrailTopicType,
  GuardrailWordPolicyAction,
  GuardrailManagedWordType,
  InvocationType,
  CreationMode,
  PromptType,
  Source,
  Type,
  OrchestrationTrace,
  PostProcessingTrace,
  PreProcessingTrace,
  Trace,
  ResponseStream,
  MemoryType,
  Memory,
  ExternalSourceType,
  QueryTransformationType,
  RetrieveAndGenerateType,
  GuadrailAction,
  RetrievalFilter,
  ActionGroupInvocationInputFilterSensitiveLog,
  ActionGroupInvocationOutputFilterSensitiveLog,
  FlowInputContentFilterSensitiveLog,
  FlowInputFilterSensitiveLog,
  InvokeFlowRequestFilterSensitiveLog,
  FlowCompletionEventFilterSensitiveLog,
  FlowOutputEventFilterSensitiveLog,
  FlowResponseStreamFilterSensitiveLog,
  InvokeFlowResponseFilterSensitiveLog,
  ByteContentFileFilterSensitiveLog,
  FileSourceFilterSensitiveLog,
  InputFileFilterSensitiveLog,
  ApiResultFilterSensitiveLog,
  InvocationResultMemberFilterSensitiveLog,
  TextResponsePartFilterSensitiveLog,
  GeneratedResponsePartFilterSensitiveLog,
  RetrievalResultContentFilterSensitiveLog,
  RetrievalResultLocationFilterSensitiveLog,
  RetrievedReferenceFilterSensitiveLog,
  CitationFilterSensitiveLog,
  AttributionFilterSensitiveLog,
  PayloadPartFilterSensitiveLog,
  OutputFileFilterSensitiveLog,
  FilePartFilterSensitiveLog,
  ApiInvocationInputFilterSensitiveLog,
  InvocationInputMemberFilterSensitiveLog,
  ReturnControlPayloadFilterSensitiveLog,
  FailureTraceFilterSensitiveLog,
  GuardrailContentFilterFilterSensitiveLog,
  GuardrailContentPolicyAssessmentFilterSensitiveLog,
  GuardrailPiiEntityFilterFilterSensitiveLog,
  GuardrailRegexFilterFilterSensitiveLog,
  GuardrailSensitiveInformationPolicyAssessmentFilterSensitiveLog,
  GuardrailTopicFilterSensitiveLog,
  GuardrailTopicPolicyAssessmentFilterSensitiveLog,
  GuardrailCustomWordFilterSensitiveLog,
  GuardrailManagedWordFilterSensitiveLog,
  GuardrailWordPolicyAssessmentFilterSensitiveLog,
  GuardrailAssessmentFilterSensitiveLog,
  GuardrailTraceFilterSensitiveLog,
  KnowledgeBaseLookupInputFilterSensitiveLog,
  InvocationInputFilterSensitiveLog,
  ModelInvocationInputFilterSensitiveLog,
  UsageFilterSensitiveLog,
  MetadataFilterSensitiveLog,
  RawResponseFilterSensitiveLog,
  OrchestrationModelInvocationOutputFilterSensitiveLog,
  FinalResponseFilterSensitiveLog,
  KnowledgeBaseLookupOutputFilterSensitiveLog,
  RepromptResponseFilterSensitiveLog,
  ObservationFilterSensitiveLog,
  RationaleFilterSensitiveLog,
  OrchestrationTraceFilterSensitiveLog,
  PostProcessingParsedResponseFilterSensitiveLog,
  PostProcessingModelInvocationOutputFilterSensitiveLog,
  PostProcessingTraceFilterSensitiveLog,
  PreProcessingParsedResponseFilterSensitiveLog,
  PreProcessingModelInvocationOutputFilterSensitiveLog,
  PreProcessingTraceFilterSensitiveLog,
  TraceFilterSensitiveLog,
  TracePartFilterSensitiveLog,
  ResponseStreamFilterSensitiveLog,
  InvokeAgentResponseFilterSensitiveLog,
  RetrieveAndGenerateInputFilterSensitiveLog,
  PromptTemplateFilterSensitiveLog,
  ExternalSourcesGenerationConfigurationFilterSensitiveLog,
  ByteContentDocFilterSensitiveLog,
  ExternalSourceFilterSensitiveLog,
  ExternalSourcesRetrieveAndGenerateConfigurationFilterSensitiveLog,
  GenerationConfigurationFilterSensitiveLog,
  OrchestrationConfigurationFilterSensitiveLog,
  RetrieveAndGenerateOutputFilterSensitiveLog,
  RetrieveAndGenerateResponseFilterSensitiveLog,
  KnowledgeBaseQueryFilterSensitiveLog,
  KnowledgeBaseRetrievalResultFilterSensitiveLog,
  RetrieveResponseFilterSensitiveLog,
  RetrievalFilterFilterSensitiveLog,
  KnowledgeBaseVectorSearchConfigurationFilterSensitiveLog,
  KnowledgeBaseRetrievalConfigurationFilterSensitiveLog,
  KnowledgeBaseConfigurationFilterSensitiveLog,
  KnowledgeBaseRetrieveAndGenerateConfigurationFilterSensitiveLog,
  RetrieveRequestFilterSensitiveLog,
  RetrieveAndGenerateConfigurationFilterSensitiveLog,
  RetrieveAndGenerateRequestFilterSensitiveLog,
  SessionStateFilterSensitiveLog,
  InvokeAgentRequestFilterSensitiveLog
});
